lib/InnerTransRel.ml

```diff
@@ -109,4 +125,18 @@ module Make (O : MySet.OrderedType) :
   let to_transitive_rel t = Rel.transitive_closure t.rel
 
   let to_implicitely_transitive_rel t = t.rel
+
+  let to_map t = Rel.M.to_map t.rel
+
+  let inverse {input; rel; output} =
+    let input = output
+    and output = input
+    and rel = Rel.inverse rel in
+    { input; rel; output }
+
+  let domain { rel } = Rel.domain rel
+  let codomain { rel } = Rel.codomain rel
+  let filter_imp_trans pred t =
+    let rel = Rel.filter pred t.rel in
+    { t with rel }
```

Comment:
Hi @HadrienRenaud. `filter_imp_trans`  probably does not performs what is intended.

Consider the implicitly transitive relation r consisting in the two pairs `{a->b,b->c}`, which thus represents `r+ = r | {a -> c}`. Also assume a predicate `is_red` , where `a` and `c` are red, bit not b, and then the predicate `pred` on pairs `pred(x->y) = is_red(x) & is_red(y)`.
  + Then on the one hand,  `filter_imp_trans pred r` is the empty relation,
  + While, on the other hand `filter r+` is `{a->c}`.